eth/catalyst/queue.go

```diff
@@ -34,11 +35,47 @@ const maxTrackedPayloads = 10
 // latest one; but have a slight wiggle room for non-ideal conditions.
 const maxTrackedHeaders = 10
 
+// payload wraps the execution payload generated by miner
```

Comment:
Perhaps expanding it would make the doc a bit more useful:

```
// payload wraps the miner's block production channel, allowing the mined block
// to be retrieved later upon the GetPayload engine API call.
```